### PR TITLE
Update filenames

### PR DESCRIPTION
**Pull request by srajiang**
### Type of change (place an x in the [ ] that applies)

- [ ] New sample
- [ ] New feature
- [ ] Bug fix
- [ ] Documentation
- [x] improvement

### Summary

This PR updates function filenames `mod.ts` --> `handler.ts`, and updates `definition.ts` --> `mod.ts` for this app's 3 workflows. Since the function definition is what is primarily being exported out of the function directories for use in workflows files and thus make sense as the default export out of the directory. This was @shapirone's suggestion 🙇 

@misscoded Adding you as reviewer, since we talked about this as a possibility, and I want to make sure there's visibility that `handler.ts` would become a general go-to standard for the file that contains SlackFunction. 

### Requirements (place an x in each [ ] that applies)

- [x] I’ve checked my submission against the Samples Checklist to ensure it complies with all standards
- [x] I have ensured the changes I am contributing align with existing patterns and have tested and linted my code
- [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct)

### CODE REVIEW

File: functions/create_draft/handler.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+      return { error: draftPostErrorMsg };
+    }
+
+    // 3. Update the draft record with the successful sent drafts timestamp
+    const putResp2 = await client.apps.datastore.put<
+      typeof DraftDatastore.definition
+    >({
+      datastore: DraftDatastore.name,
+      // @ts-expect-error expecting fix in future SDK release
```
Comment (shapirone) [inline functions/create_draft/handler.ts:89]: What's the type system complaining about here?

**Review by shapirone — APPROVED** [review]
I think this is an improvement, I do think there still may be a risk for confusion around using `handler.ts`. From what I know we pretty consistently refer to interactivity handlers, but I'm not sure what the common term is for the default handler that we're putting in `handler.ts`

File: functions/create_draft/handler.ts (hunk 1 above)
Comment (srajiang) [reply functions/create_draft/handler.ts:89]: Missing a datastore item field. 

Screenshot 2023-01-13 at 9 50 20 AM

For example here,`message_ts` is a property that's in the datastore definition but isn't available yet (it's added in a later part of the code).

**Comment by srajiang** [comment]
@shapirone - Yeah, it's possible. I think the "default handler" or the "primary handler" is what I have seen people use. That's why I am using `handler.ts` and `interactivity_handler.ts` as filenames to hint at their relationship. If `SlackFunction` could be aliased or even directly exported from the module as `SlackFunctionHandler`, would that make things more clear/consistent for you?

An alternative maybe is `function.ts`, but I think that could also be confusing for devs for other reasons.

**Comment by misscoded** [comment]
What I'm still not entirely sold on is the use of `mod.ts`. If this file is to just contain the definition, why is it not named `definition.ts`? Doing so would gently discourage developers from putting anything other than the `DefineFunction` in there, which is what we're attempting to guide them towards -- keeping things separate and their files clean/tidy.

Circling back to the handler/function, with `definition.ts`, which is the better, more intuitive pair of the two: `definition.ts` and `handler.ts` OR `definition.ts` and `function.ts`?

**Comment by srajiang** [comment]
```
/ functions                              # function is already in the root, so we can avoid naming anything "*_function" below this
   / create_approval                     # named of the function
      / definition.ts                    # to me, this can be either mod.ts or definition.ts, and leaning definition.ts for reasons Alissa mentioned
      / handler.ts                       # prefer this over function.ts because of redundancy point 
      / handler_interaction.ts (opt.)    # this would be how I'd break out any additional handlers. the combo of `function.ts` and `function_interaction_handler.ts` to me as an alternative feels cumbersome
```

Pulling all this feedback together, ☝️  is what makes the most sense to me. 👍  👎  additional thoughts @misscoded @shapirone?

**Comment by misscoded** [comment]
LGTM, with the exception of `handler_interaction.ts`. I defer to you both on that point.

**Comment by shapirone** [comment]
> What I'm still not entirely sold on is the use of mod.ts. If this file is to just contain the definition, why is it not named definition.ts? Doing so would gently discourage developers from putting anything other than the DefineFunction in there, which is what we're attempting to guide them towards -- keeping things separate and their files clean/tidy.

The [Deno Style guide](https://deno.land/manual@v1.29.4/references/contributing/style_guide#do-not-use-the-filename-indextsindexjs) denotes the `mod.ts` file as the default entry point of the directory. IMO, we should stick to that convention. Assuming there is agreement there, the immediate  question to me from that decision is  _"Do we need a default entry point?"_ My opinion is that we should encourage the use of `mod.ts` as a way to export anything that will be useful outside of the function's directory. This could include the function definition, the callback ID, any types that we construct, maybe some of the block helpers we choose not to make available at a higher level, etc.

Okay, so there is a use for a `mod.ts` file, but should that be where the definition is defined or should we have the definition in it's own `definition.ts` file and use `mod.ts` to simply re-export modules? This part to me is very much based on user preference. Personally I tend to avoid abstracting functionality into it's own file until it becomes burdensome to keep it in the same file so I would probably leave it in the `mod.ts` file. This is primarily due to a preference to limit the amount of separate files that I work in, but this is a weakly held position, especially considering this is a sample for others to reference and/or build from. Since we're attempting to guide them to keep things separate and their files clean/tidy let's use `definition.ts`.

> Circling back to the handler/function, with definition.ts, which is the better, more intuitive pair of the two: definition.ts and handler.ts OR definition.ts and function.ts?

I'm in favor of using `handler.ts` as I'm worried that the term function is very overloaded in this space. We're creating our own function entity that is layered on top of native functions. Knowing that all of the files we're discussing exists under a `functions/` directory, I imagine it will be pretty easy to get confused and tripped up in that space. We already have this problem internally 😅 Unfortunately, I'm not sure we have a common term for the default handler, so i think just `handler.ts` makes the most sense here.

> LGTM, with the exception of handler_interaction.ts. I defer to you both on that point.

I would like us to call this `interactivity_handler.ts` as that's how we commonly refer to it and it'd be nice to be consistent. I couldn't find this term in the public API docs, but it's in every interactivity handler doc in [the repo](https://github.com/slackapi/deno-slack-sdk/blob/main/docs/functions-action-handlers.md#requirements).


TLDR:

1. We should use `mod.ts` as the default entry point to each function's directory so devs can import everything they'll need from that function via `/functions/*/mod.ts`
2. It's fine to separate the `DefineFunction` call into it's own `definition.ts` file if we import it into `mod.ts`
3. Stick to using `handler.ts` instead of `function.ts`
4. Switch from `handler_interaction` to `interactivity_handler`

Example:
```
/ functions                              
   / create_approval               
      / mod.ts                          # the default entrypoint to the create_approval function that devs will use to import from
      / definition.ts                    # fine to use definition file for where to define the definition... but should import into mod.ts
      / handler.ts                       # also prefer this over function.ts because of redundancy point
      / interactivity_handler.ts (opt.)    # change from interaction_handler
```

**Comment by srajiang** [comment]
Thanks both. This is great - I changed around the order of `interactivity_handler.ts` to `handler_interactivity.ts` because it was nicer from a file sorting standpoint, but I agree with you that `interactivity_handler` is more consistent with how we generally refer to things. 

I will go ahead and make those changes accordingly :) Just in time as we're going to demo this sample structure to wider DevRel tomorrow and it'll inform future samples at least for now until we revise again.

**Comment by misscoded** [comment]
And so it shall be! We'll codify this in the Samples' Standards docs and apply it to subsequent reviews going forward for more complex applications. Thanks for the thorough discussion!
